beancount.el

```diff
@@ -542,6 +548,19 @@ will allow to align all numbers."
      ((or (looking-at-p beancount-timestamp-indent-regexp)
           (looking-at-p beancount-directive-indent-regexp))
       0)
+     ;; Metadata lines should have an additional level of indentation
+     ;; if the previous line is a directive (timestamped or not) or a
+     ;; posting. Otherwise they should have the same level of
+     ;; indentation as the previous line.
+     ((looking-at-p beancount-metadata-indent-regexp)
+      (if (/= (forward-line -1) 0)
+          0
+        (if (or (looking-at-p beancount-transaction-regexp)
+                (looking-at-p beancount-directive-regexp)
+                (looking-at-p beancount-timestamped-directive-regexp)
+                (looking-at-p beancount-posting-regexp))
+            (+ (current-indentation) beancount-transaction-indent)
+          (current-indentation))))
```

Comment:
Your suggestion does not work since postings can have multiple lines of metadata following them, for example, the test case that I extended:

```
2019-01-01 * \"Example\"
  transaction: metadata
  #foo
  ^bar
  also_transaction: metadata
  Expenses:Example                              1.00 USD
    posting: metadata
    also_posting: metadata
  Assets:Checking                              -1.00 USD
    second_posting: metadata
```